### DISCUSSION TITLE
Android fails to connect on MetaMask or Trust, iOS is fine

### DISCUSSION THREAD

**Discussion by fvongraf** [WalletConnect/walletconnect-monorepo#2801] [Developer Support]
We have been using WC V1 from within an iframe running React, and it has been working fine across Android and iPhones, and multiple mobile browsers.  When switching to WC V2 we are still able to have iPhones connect and transact, but Android can't even connect.  The console is showing the error below.

One of the error messages we're am getting on Android: failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://verify.walletconnect.com/') does not match the recipient window's origin ('https://c****.myoso.io/'). (anonymous) @ ws.ts:197

Even though HTML is in the document being served, this error is being caused by the fact that this is using an iFrame for our web app. 

Any help on how we can get wallet connections working again would be greatly appreciated.

**Comment by crypblizz8** [comment]
Hey @fvongraf, thanks for your question. Does this documentation here help you solve the issue:
https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps/dapp-checklist#common-errors-and-fixes

**Reply by fvongraf** [reply]
Unfortunately not, we are not a DAPP or Wallet that is looking to be connected, but rather a web app that is using WalletConnect to connect to wallets.  Within our web page we are able to connect wallets when using an iPhone but when going to the identical web page on Android it fails to connect using WC and has these errors.

This only started when we went to using WC V2.
